reproduce.sh

```diff
@@ -31,13 +31,33 @@ while [ "$1" != "" ]; do
   shift
 done
 
+# Later checks will set this to false if dependencies are missing.
+dependencies_installed="true"
+
+# Create the config directory if needed and check the ClusterFuzz version.
 mkdir -p $CLUSTERFUZZ_CONFIG_DIR
-if [ ! -d $ROOT_DIRECTORY/ENV ] || ([ $additional_deps_args ] && [ ! -d $ROOT_DIRECTORY/local/bin/android-sdk ]); then
+current_version=1
+version_file=$CLUSTERFUZZ_CONFIG_DIR/version
+if [ ! -e $version_file ] || [ "$(cat $version_file)" != "$current_version" ]; then
+  dependencies_installed="false"
+fi
+
+# Check if we have a valid pipenv.
+if ! pipenv graph 2>&1 > /dev/null; then
+  dependencies_installed="false"
+fi
+
+# Check to see if we need to install the android emulator.
+if [ $additional_deps_args ] && [ ! -d $ROOT_DIRECTORY/local/bin/android-sdk ]; then
+  dependencies_installed="false"
+fi
+
+if [ "$dependencies_installed" != "true" ]; then
   echo "Running first time setup. This may take a while, but is only required once."
   echo "You may see several password prompts to install required packages."
   sleep 5
-  $ROOT_DIRECTORY/local/install_deps.bash --only-reproduce $additional_deps_args || { rm -rf $ROOT_DIRECTORY/ENV && exit 1; }
+  PY3=1 $ROOT_DIRECTORY/local/install_deps.bash --only-reproduce $additional_deps_args || { exit 1; }
+  echo -n $version > $CLUSTERFUZZ_CONFIG_DIR/version
 fi
 
-source ENV/bin/activate
-python $ROOT_DIRECTORY/butler.py reproduce $original_args
+pipenv run python $ROOT_DIRECTORY/butler.py reproduce $original_args
```

Comment:
We should run `pipenv sync --dev` before here to update any dependencies that might have changed.